x/msgfee/decorator_test.go

```diff
@@ -0,0 +1,70 @@
+package msgfee
+
+import (
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/x"
+)
+
+func TestFeeDecorator(t *testing.T) {
+	handler := helpers.CountingHandler()
+	decorator := NewFeeDecorator()
+	bucket := NewMsgFeeBucket()
+	db := store.MemStore()
+
+	_, err := bucket.Create(db, &MsgFee{
+		MsgPath: "foo/bar",
+		Fee:     coin.NewCoin(0, 1234, "DOGE"),
+	})
+	if err != nil {
+		t.Fatalf("cannot create a transaction fee: %s", err)
+	}
+
+	tx := &txMock{
+		msg: &msgMock{path: "foo/bar"},
+	}
+	if res, err := decorator.Check(nil, db, tx, handler); err != nil {
+		t.Fatalf("check failed: %s", err)
+	} else if !res.RequiredFee.Equals(coin.NewCoin(0, 1234, "DOGE")) {
+		t.Fatalf("unexpected check fee: %v", res.RequiredFee)
+	}
+
+	if c := handler.GetCount(); c != 1 {
+		t.Fatalf("want count=1, got %d", c)
+	}
+
+	if res, err := decorator.Deliver(nil, db, tx, handler); err != nil {
+		t.Fatalf("check failed: %s", err)
+	} else if !res.RequiredFee.Equals(coin.NewCoin(0, 1234, "DOGE")) {
+		t.Fatalf("unexpected deliver fee: %v", res.RequiredFee)
+	}
+
+	if c := handler.GetCount(); c != 2 {
+		t.Fatalf("want count=2, got %d", c)
+	}
+}
+
+var helpers x.TestHelpers
```

Comment:
Do you want to extend x.TestHelpers with these functionalities?

Reply:
I would like to, but preferably after the https://github.com/iov-one/weave/issues/372